### DISCUSSION TITLE
When can we expect OpenAPI 3.1 support?

### DISCUSSION THREAD

**Discussion by Stokestack** [stoplightio/spectral#1653] [Q&A] [answered]
There are some very useful additions in 3.1, such as the ability to add a description alongside a use of a schema reference.

I did a search for OpenAPI 3.1 here and got nothing... which is concerning. I'm using Stoplight Studio, which in turn uses Spectral to validate its documents.

**Comment by P0lip** [comment, marked as answer]
The initial support for OAS 3.1 has already landed in the V6 version, which is still considered an alpha.
We're planning to roll out the stable release of V6 soon, hopefully in June/July.
In general, V6 is stable enough to use. The only reason it's still in the alpha stage is the fact we have a few remaining breaking changes to introduce.

**Reply by Stokestack** [reply]
Thanks for the info.

**Reply by mnaumanali94** [reply]
Oh also, if you're using Stoplight studio we have already introduced alpha support in there.

**Reply by Stokestack** [reply]
Thanks. Yes, I added the 3.1 header to my doc and immediately some things were no longer flagged as errors.

Unfortunately some important additions are still missing from the form view, such as a Links collection (actually I think that was already a fundamental part of OpenAPI before 3.1), and the ability to provide a description alongside a use of a reference (a baffling omission from the earlier OpenAPI spec).

**Reply by alex-feel** [reply]
Is there a timeline when you plan to update [Spectral Linter Action](https://github.com/stoplightio/spectral-action) to use Spectral v6?

**Reply by benjamin-mogensen** [reply]
For me validation with Spectral breaks when I set the version `openapi: 3.1.0` and I basically get this:
```
  {
    code: 'oas3-schema',
    message: '`openapi` property must match pattern `^3\\.0\\.\\d(-.+)?$`.',
    path: [ 'openapi' ],
    severity: 0,
    range: { start: [Object], end: [Object] }
  }
```
I am running on 6.0.0-alpha3 so hope this will be fixed in the final v6
